pkg/prometheus/operator.go

```diff
@@ -284,3 +285,21 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 
 	return &pStatus, nil
 }
+
+func MakeSelectorLabels(p monitoringv1.PrometheusInterface) map[string]string {
+	name := p.GetObjectMeta().GetName()
+	labelMap := map[string]string{
+		"app.kubernetes.io/managed-by": "prometheus-operator",
+		"app.kubernetes.io/instance":   name,
+		PrometheusNameLabelName:        name,
+	}
+	switch p.(type) {
+	case *monitoringv1.Prometheus:
+		labelMap["app.kubernetes.io/name"] = "prometheus"
+		labelMap["prometheus"] = name
+	case *monitoringv1alpha1.PrometheusAgent:
+		labelMap["app.kubernetes.io/name"] = "prometheus-agent"
+	default:
+	}
+	return labelMap
+}
```

Comment:
Kind of annoying that Prometheus server has an extra label when compared to Prometheus agent 😬 

I've done a type switch here to properly add the individual labels, let me know if you prefer a different strategy

Reply:
We could also have an extra method to the Interface itself 🤔

Reply:
I find it simpler to have dedicated functions in pkg/prometheus/server and pkg/prometheus/agent in this case. Moving logic to pkg/apis/monitoring increases the surface of the API package (and can create issues like #6203).